py/server.py

```diff
@@ -730,6 +738,31 @@ def post(self):
         self.wrap_func(self, args)
 
 
+class DataHandler(BaseHandler):
+    def initialize(self, app):
+        self.state = app.state
+
+    @staticmethod
+    def wrap_func(handler, args):
+        eid = extract_eid(args)
+
+        if 'win' in args and args['win'] is not None:
```

Comment:
`and args['win'] in handler.state[eid]['jsons']`?

Reply:
I should put this as an assert - errors are passed from the server back to the user.

Reply:
Even better!